web/src/components/PersonalInFo/PersonalInfoForm.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import { useSubmitPersonalInfo } from '../../hooks/useSubmitPersonalInfo.js';
+import styles from './PersonalInfo.module.css';
+
+const PersonalInfoForm = () => {
+  const [personalData, setPersonalData] = useState({
+    fullName: '',
+    email: '',
+    phone: '',
+    github: '',
+    linkedin: '',
+    portfolio: '',
+  });
+
+  const { submitPersonalInfo, loading, error, successMessage } =
+    useSubmitPersonalInfo();
+
+  const handleChange = (e) => {
+    setPersonalData({ ...personalData, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    submitPersonalInfo(personalData);
+
+    setPersonalData({
+      fullName: '',
+      email: '',
+      phone: '',
+      github: '',
+      linkedin: '',
+      portfolio: '',
+    });
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className={styles.form}>
+      <label>
+        FullName:
+        <input
+          type="text"
+          name="fullName"
+          placeholder="Full Name"
+          value={personalData.fullName}
+          onChange={handleChange}
+          className={styles.input}
+          required
+        />
+      </label>
+
+      <label>
+        Email:
+        <input
+          type="email"
+          name="email"
+          placeholder="Email"
+          value={personalData.email}
+          onChange={handleChange}
+          className={styles.input}
+          required
+        />
+      </label>
+
+      <label>
+        Phone:
+        <input
+          type="tel"
+          name="phone"
+          placeholder="Phone"
+          value={personalData.phone}
+          onChange={handleChange}
+          className={styles.input}
+          required
+        />
+      </label>
+
+      <label>
+        GitHub:
+        <input
+          type="url"
+          name="github"
+          placeholder="GitHub URL"
+          value={personalData.github}
+          onChange={handleChange}
+          className={styles.input}
+          required
+        />
+      </label>
+
+      <label>
+        LinkedIn:
+        <input
+          type="url"
+          name="linkedin"
+          placeholder="LinkedIn URL"
+          value={personalData.linkedin}
+          onChange={handleChange}
+          className={styles.input}
+          required
+        />
+      </label>
+
+      <label>
+        Portfolio:
+        <input
+          type="url"
+          name="portfolio"
+          placeholder="Portfolio URL"
+          value={personalData.portfolio}
+          onChange={handleChange}
+          className={styles.input}
+          required
+        />
+      </label>
+
+      <button type="submit" disabled={loading} className={styles.button}>
+        {loading ? 'Saving...' : 'Save'}
+      </button>
+
+      {error && <p className={styles.error}>{error}</p>}
```

Comment:
this error class does not exist in the css we should add a basic version in, perhaps just changing the color to red

![image](https://github.com/user-attachments/assets/aad9ead0-07ff-462e-b196-ffc82c7a150f)